src/node/internal/internal_fs_sync.ts

```diff
@@ -0,0 +1,1066 @@
+  atime = getDate(atime);
+  mtime = getDate(mtime);
+  cffs.setLastModified(normalizePath(path), mtime, { followSymlinks: false });
+}
+
+export function linkSync(existingPath: FilePath, newPath: FilePath): void {
+  cffs.link(normalizePath(newPath), normalizePath(existingPath), {
+    symbolic: false,
+  });
+}
+
+export type StatOptions = {
+  bigint?: boolean;
+  throwIfNoEntry?: boolean;
+};
+
+export function lstatSync(
+  path: FilePath,
+  options: StatOptions = { bigint: false, throwIfNoEntry: true }
+): Stats | undefined {
+  validateObject(options, 'options');
+  const { bigint = false, throwIfNoEntry = true } = options;
+  validateBoolean(bigint, 'options.bigint');
+  validateBoolean(throwIfNoEntry, 'options.throwIfNoEntry');
+  const stat = cffs.stat(normalizePath(path), { followSymlinks: false });
+  if (stat == null) {
+    if (throwIfNoEntry) {
+      throw new Error('file not found'); // not found or not accessible.
+    }
+    return undefined;
+  }
+  return new Stats(kBadge, stat, { bigint });
+}
+
+export type MkdirSyncOptions = {
+  recursive?: boolean;
+  mode?: string | number;
+};
+
+export function mkdirSync(
+  path: FilePath,
+  options: number | MkdirSyncOptions = {}
+): string | undefined {
+  const { recursive = false, mode = 0o777 } = ((): MkdirSyncOptions => {
+    if (typeof options === 'number') {
+      return { mode: options };
+    } else {
+      validateObject(options, 'options');
+      return options;
+    }
+  })();
+
+  validateBoolean(recursive, 'options.recursive');
+
+  // We don't implement the mode option in any meaningful way. We just validate it.
+  parseFileMode(mode, 'mode');
+
+  return cffs.mkdir(normalizePath(path), { recursive, tmp: false });
+}
+
+export type MkdirTempSyncOptions = {
+  encoding?: string;
+};
+
+export function mkdtempSync(
+  prefix: FilePath,
+  options: MkdirTempSyncOptions = {}
+): string {
+  validateObject(options, 'options');
+  const { encoding = 'utf8' } = options;
+  validateEncoding(encoding, 'options.encoding');
+  prefix = normalizePath(prefix, encoding);
+  const ret = cffs.mkdir(normalizePath(prefix), {
+    recursive: false,
+    tmp: true,
+  });
+  if (ret === undefined) {
+    // TODO(node:fs): Need proper error code here
+    throw new Error('failed to create temporary directory');
+  }
+  return ret;
+}
+
+export type OpenDirOptions = {
+  encoding?: string;
+  bufferSize?: number;
+  recursive?: boolean;
+};
+
+export function opendirSync(
+  path: FilePath,
+  options: OpenDirOptions = {
+    encoding: 'utf8',
+    bufferSize: 32,
+    recursive: false,
+  }
+): Dir {
+  validateObject(options, 'options');
+  const { encoding = 'utf8', bufferSize = 32, recursive = false } = options;
+  validateEncoding(encoding, 'options.encoding');
+  validateUint32(bufferSize, 'options.bufferSize');
+  validateBoolean(recursive, 'options.recursive');
+  path = normalizePath(path);
+  throw new Error('Not implemented');
+}
+
+export function openSync(
```

Comment:
What about `openAsBlob`? Again comment or placeholder might help.

Reply:
Planning to open that as a separate PR